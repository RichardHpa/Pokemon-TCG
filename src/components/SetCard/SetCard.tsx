import { Card } from 'components/Card'
import { Paragraph } from 'components/Paragraph'
import { Heading } from 'components/Heading'

import type { Set } from 'types/fixtures/set'

interface SetCardProps
  extends Pick<Set, 'name' | 'series' | 'releaseDate' | 'total' | 'printedTotal'> {
  image: string
}

export const SetCard = ({
  image,
  name,
  series,
  releaseDate,
  total,
  printedTotal,
}: SetCardProps) => {
  const secretTotal = total - printedTotal
  return (
    <Card>
      <div className='flex flex-col h-full'>
        <div className='flex justify-center mb-4'>
          <img className='w-full max-w-xs' src={image} alt={`${name} logo`} />
        </div>

        <div className='mt-auto'>
          <Heading level='4'>{name}</Heading>

          <Paragraph>
            {series} - {releaseDate}
          </Paragraph>
          <Paragraph>
            {total} Cards ({secretTotal} Secret Cards)
          </Paragraph>
        </div>
      </div>
    </Card>
  )
}
