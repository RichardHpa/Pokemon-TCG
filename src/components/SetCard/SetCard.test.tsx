import { render, screen } from '@testing-library/react'

import { set } from 'fixtures/sets'

import { SetCard } from './SetCard'

const props = {
  name: set.name,
  releaseDate: set.releaseDate,
  series: set.series,
  total: set.total,
  printedTotal: set.printedTotal,
  image: set.images.logo,
}

describe('SetCard', () => {
  test('renders SetCard', () => {
    render(<SetCard {...props} />)
    expect(screen.getByText(set.name)).toBeInTheDocument()
    expect(screen.getByText(`${set.series} - ${set.releaseDate}`)).toBeInTheDocument()
    expect(screen.getByText(`${set.total} cards`)).toBeInTheDocument()
  })

  test('renders SetCard snapshot', () => {
    const { container } = render(<SetCard {...props} />)
    expect(container).toMatchSnapshot()
  })
})
