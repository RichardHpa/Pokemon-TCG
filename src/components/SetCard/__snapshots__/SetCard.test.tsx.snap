// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetCard renders SetCard snapshot 1`] = `
<div>
  <div
    class="bg-white rounded-lg border border-gray-200 shadow-md dark:bg-gray-800 dark:border-gray-700 p-4 h-full"
  >
    <div
      class="flex flex-col h-full"
    >
      <div
        class="flex justify-center mb-4"
      >
        <img
          alt="Lost Origin logo"
          class="w-full max-w-xs"
          src="https://images.pokemontcg.io/swsh11/logo.png"
        />
      </div>
      <div
        class="mt-auto"
      >
        <h4
          class="font-bold tracking-tight text-gray-900 dark:text-white mb-2 text-2xl"
        >
          Lost Origin
        </h4>
        <p
          class="mb-3 text-gray-500 dark:text-gray-400"
        >
          Sword & Shield
           - 
          2022/09/09
        </p>
        <p
          class="mb-3 text-gray-500 dark:text-gray-400"
        >
          217
           Cards (
          21
           Secret Cards)
        </p>
      </div>
    </div>
  </div>
</div>
`;
